oomact_ros/src/CalibrationNode.cpp

```diff
@@ -0,0 +1,119 @@
+#include <rosbag/bag.h>
+#include <ros/ros.h>
+
+#include <aslam/calibration/ros/RosInputProvider.h>
+#include <aslam/calibration/calibrator/CalibratorI.hpp>
+#include <aslam/calibration/input/InputProviderI.h>
+#include <aslam/calibration/data/ObservationManagerI.h>
+#include <aslam/calibration/model/FrameGraphModel.h>
+#include <aslam/calibration/model/Model.h>
+#include <aslam/calibration/model/PoseTrajectory.h>
+#include <aslam/calibration/model/fragments/So3R3Trajectory.h>
+#include <aslam/calibration/model/sensors/PoseSensor.hpp>
+#include <aslam/calibration/tools/SmartPointerTools.h>
+
+#include <sm/boost/null_deleter.hpp>
+#include <sm/BoostPropertyTree.hpp>
+#include <sm/value_store/LayeredValueStore.hpp>
+#include <sm/value_store/PrefixedValueStore.hpp>
+
+using namespace aslam;
+namespace cal = aslam::calibration;
+
+cal::ValueStoreRef valueStoreFromFile(std::string file_path,
+                                 sm::BoostPropertyTree *bpt_ptr = nullptr) {
+  sm::BoostPropertyTree bpt;
+  if (bpt_ptr) {
+    *bpt_ptr = bpt;
+  }
+  bpt.load(file_path);
+  return cal::ValueStoreRef(bpt);
+}
+
+int main(int argc, char **argv) {
+  ros::init(argc, argv, "CalibrationNode");
+
+  ros::NodeHandle nh, nh_private("~");
+
+  ROS_INFO("Started node");
+
+  google::ParseCommandLineFlags(&argc, &argv, true);
+  google::InitGoogleLogging(argv[0]);
+  google::SetStderrLogging(FLAGS_v > 0 ? google::INFO : google::WARNING);
+  google::InstallFailureSignalHandler();
+
+  ROS_INFO("Loading config file");
+
+  std::string config_file;
+  if (!nh_private.getParam("config_file", config_file)) {
+    ROS_FATAL("Could not find config_file parameter, exiting");
+    exit(EXIT_FAILURE);
+  }
+
+  sm::BoostPropertyTree bpt;
+  bpt.load(config_file);
+
+  cal::ValueStoreRef vs(bpt);
```

Comment:
The last three lines could now (since https://github.com/ethz-asl/Schweizer-Messer/pull/192) be:
`auto vs = cal::ValueStoreRef::fromFile(config_file);`